File: 02a.go
package main

import (
    "os"
    "log"
    "bufio"
    "strconv"
    "strings"
    "math"
    "fmt"
)

func main() {
    file, err := os.Open("02.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()

    var offsetHorizontal int = 0
    var offsetVertical int = 0
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        lineData := strings.Split(scanner.Text(), " ")
        direction := lineData[0]
        amount, err := strconv.Atoi(lineData[1])
        if err != nil {
            log.Fatal(err)
        }

        if (direction == "forward") {
            offsetHorizontal += amount
        } else if (direction == "down") {
            offsetVertical += amount
        } else if (direction == "up") {
            offsetVertical -= amount
        } else {
            log.Fatal("Wrong direction in line ", lineData)
        }
    }

    fmt.Println("horizontal position: ", offsetHorizontal)
    fmt.Println("vertical position: ", offsetVertical)
    fmt.Println("result: ", int(math.Abs(float64(offsetHorizontal * offsetVertical))))
}

File: python/02a.py
inputFile = open('02.txt', 'r')
lines = inputFile.readlines()

offsetHorizontal = 0
offsetVertical = 0
for line in lines:
    data = line.split()
    direction = data[0].strip()
    amount = int(data[1].strip())
    if direction == 'forward':
        offsetHorizontal += amount
    elif direction == 'down':
        offsetVertical += amount
    elif direction == 'up':
        offsetVertical -= amount
    else:
        print(f'wrong direction in line {line}')

print(f'horizontal position: {offsetHorizontal}')
print(f'vertical position: {offsetVertical}')
print(abs(offsetHorizontal * offsetVertical))
